org/mozilla/jss/nss/SSL.c

```diff
@@ -842,6 +891,27 @@ Java_org_mozilla_jss_nss_SSL_RemoveCallbacks(JNIEnv *env, jclass clazz,
     SSL_AuthCertificateHook(real_fd, NULL, NULL);
 }
 
+JNIEXPORT jint JNICALL
+Java_org_mozilla_jss_nss_SSL_EnableHandshakeCallback(JNIEnv *env, jclass clazz,
+    jobject fd)
+{
+    PRFileDesc *real_fd = NULL;
+    jobject fd_ref = NULL;
+
+    PR_ASSERT(env != NULL && fd != NULL);
```

Comment:
Question, since this block of code seems to happen in a bunch of methods, would having a convenience method be of use?

Reply:
A lot of code uses this pattern. I don't think you can use a convenience method because in debug mode, it aborts. It'd be a candidate for a macro in the future (because we need to return from this method), but a lot of existing code also needs to be replaced. I think we need to replace it with a macro that checks these for `NULL` and returns `NullPointerException`.

Reply:
They could also be removed, as the JNI contract should guarantee `env` should be non-`NULL`. If `env` is `NULL`, we have no safe way of handling it other than aborting. So `env != NULL` should probably be removed from all functions and we can accept the segfault that occurs in the unlikely scenario that `env` is ever `NULL`. We're essentially doing that now as `PR_ASSERT` only executes in debug mode. 

That leaves `fd != NULL`. This (and a bunch of other cases where it is used) should be replaced by a macro that fails with NPE if/when this is `NULL`. It is up to the caller to ensure non-NULL of `fd` in this scenario.